src/core/Akka/Actor/Props.cs

```diff
@@ -353,15 +323,15 @@ public override int GetHashCode()
             SupervisorStrategy supervisorStrategy = null) where TActor : ActorBase
         {
             if (factory.Body is UnaryExpression)
-                return new DynamicProps<TActor>(factory.Compile());
+                return new Props(new FactoryConsumer<TActor>(factory.Compile()), DefaultDeploy, typeof(TActor), NoArgs);
 
-            var newExpression = factory.Body.AsInstanceOf<NewExpression>();
-            if (newExpression == null)
-                throw new ArgumentException("The create function must be a 'new T (args)' expression");
+            var newExpression = factory.Body as NewExpression
+                ?? throw new ArgumentException("The create function must be a 'new T (args)' expression");
 
             var args = newExpression.GetArguments();
+            args = args.Length > 0 ? args : NoArgs;
 
-            return new Props(new ActivatorProducer(typeof(TActor), args), DefaultDeploy, args){ SupervisorStrategy = supervisorStrategy };
+            return new Props(ActivatorProducer.Instance, DefaultDeploy, typeof(TActor), args) { SupervisorStrategy = supervisorStrategy };
```

Comment:
If we're already getting the args, would there be any benefit in also sniffing out the Ctor in `.GetArguments()`? (Can we even do that in a way that works OK with expectations of Props?)

That method was originally CopyPasta'd from OddJob, where we weren't newing up objects but rather were getting arguments for method calls. If you can actually hang on to what constructor is called (would be easy) you there are opportunities to elide penalties involved with `Activator.CreateInstance`

Reply:
Can you file an issue for this so we remember to do this as part of 1.5?

Reply:
@to11mtm We could move the logic into a custom-actor-producer and use the instance for mass creation of actors with the same props-instance, everything else will always require runtime binding or only dynamic-compile once.